cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -177,16 +175,14 @@ public final ProgramController run(final Program program, final ProgramOptions o
                             new LocalizeResource(saveHConf(hConf, File.createTempFile("hConf", ".xml", tempDir))));
       localizeResources.put("cConf.xml",
                             new LocalizeResource(saveCConf(cConf, File.createTempFile("cConf", ".xml", tempDir))));
-      File programDir = DirUtils.createTempDir(tempDir);
-      final Program copiedProgram = copyProgramJar(program, tempDir, programDir);
```

Comment:
The copying is now part of the standard contract of `AbstractProgramRuntimeService`. I just figured that after some refactoring there was bug that I forget to do the copying in the `AbstractProgramRuntimeService`. Fixed it in next commit